### PR TITLE
Add devcontainer configurations

### PR DESCRIPTION
**Pull request by mkoeppe**
### Description

Adding two devcontainer configurations
- default configuration "Docker", equivalent to using `run_docker`
- configuration "Conda".

Codespaces and VS Code allow users to select the configuration.

Previous discussion by @rebornix @rth @bamurtaugh @ryanking13 in #2602

### Checklists

- [ ] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry
- [ ] Add / update tests
- [x] Add new / update outdated documentation

### CODE REVIEW

File: .devcontainer/conda/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "Conda",
+  "image": "mcr.microsoft.com/vscode/devcontainers/base:0-bullseye",
```
Comment (bamurtaugh) [inline .devcontainer/conda/devcontainer.json:3]: We've deprecated the `vscode` org images in favor of `devcontainers` as we've developed the open Dev Container Specification (more info [here](https://github.com/microsoft/vscode-dev-containers/issues/1762)).

I'd recommend using the image in https://github.com/devcontainers/images/tree/main/src/base-debian, so something like:

```suggestion
  "image": "mcr.microsoft.com/devcontainers/base:bullseye",
```

Comment (mkoeppe) [reply .devcontainer/conda/devcontainer.json]: Thank you!

File: .devcontainer/onCreate-docker.sh [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# Do not keep running on errors
+set -e
+
+# https://pyodide.org/en/stable/development/new-packages.html#prerequisites
+pip install -e ./pyodide-build
+make -C emsdk
+make -C cpython
```
[the remark this answers is not in the document]
Comment (mkoeppe) [reply .devcontainer/onCreate-docker.sh]: Thanks, good point; I've removed it in b5de91b

**Comment by hoodmane** [comment]
Thanks @mkoeppe!

**Pull request merged**